index.html

```diff
@@ -2971,15 +2971,35 @@ <h3>Add Proof Set/Chain</h3>
 the <var>unsecuredDocument</var>.
           </li>
           <li>
+Let <var>matchingProofs</var> be an empty set.
+          </li>
+          <li>
 If <var>options</var> contains a <code>previousProof</code> attribute and that
-attribute is a string, check whether
-an element of the <var>allProofs</var> has a matching <code>id</code> attribute.
-If not, an error MUST be raised and SHOULD convey an error type of
-<a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>. If the
-<code>previousProof</code> attribute is an array, check that each element of that
-array is a string which matches the <code>id</code> attribute of an element
-of the <var>allProofs</var>. If not, an error MUST be raised and SHOULD convey
-an error type of <a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>.
+attribute is a string, add the element from <var>allProofs</var> with a matching
+<code>id</code> attribute to matchingProofs. If it does not exist, an error MUST
+be raised and SHOULD convey an error type of
+<a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>.
+If the <code>previousProof</code> attribute is an array, add an element from
+<var>allProofs</var> with an <code>id</code> attribute that matches each element
```

Comment:
I think this reads slightly better:

```suggestion
If the <code>previousProof</code> attribute is an array, add each element from
<var>allProofs</var> with an <code>id</code> attribute that matches an element
```